sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -29,5 +30,86 @@ public abstract class TokenCredential
+        /// <remarks>
+        /// Typically, the <see cref="TokenCredential"/> created by this method is for use when you have already obtained an <see cref="AccessToken"/>
+        /// from some other source and need a <see cref="TokenCredential"/> that will simply return that token. Because the static token can expire,
+        /// the delegates offer a mechanism to handle <see cref="AccessToken"/> renewal.
+        /// </remarks>
+        /// <param name="getToken">A delegate that returns an <see cref="AccessToken"/>.</param>
+        /// <param name="getTokenAsync">A delegate that returns a <see cref="ValueTask"/> of type <see cref="AccessToken"/>.</param>
+        /// <returns></returns>
+        public static TokenCredential Create(
+            Func<TokenRequestContext, CancellationToken, AccessToken> getToken,
+            Func<TokenRequestContext, CancellationToken, ValueTask<AccessToken>> getTokenAsync) => new StaticTokenCredential(getToken, getTokenAsync);
+
+        /// <summary>
+        /// Creates a static <see cref="TokenCredential"/> that accepts delegates which will produce an <see cref="AccessToken"/>.
+        /// </summary>
+        /// <remarks>
+        /// Typically, the <see cref="TokenCredential"/> created by this method is for use when you have already obtained an <see cref="AccessToken"/>
+        /// from some other source and need a <see cref="TokenCredential"/> that will simply return that token. Because the static token can expire,
+        /// the delegates offer a mechanism to handle <see cref="AccessToken"/> renewal.
+        /// </remarks>
+        /// <param name="getToken">A delegate that returns an <see cref="AccessToken"/>.</param>
+        /// <returns></returns>
+        public static TokenCredential Create(
+            Func<TokenRequestContext, CancellationToken, AccessToken> getToken) => new StaticTokenCredential(getToken);
+
+        /// <summary>
+        /// Creates a static <see cref="TokenCredential"/> that accepts delegates which will produce an <see cref="AccessToken"/>.
+        /// </summary>
+        /// <remarks>
+        /// Typically, the <see cref="TokenCredential"/> created by this method is for use when you have already obtained an <see cref="AccessToken"/>
+        /// from some other source and need a <see cref="TokenCredential"/> that will simply return that token. Because the static token can expire,
+        /// the delegates offer a mechanism to handle <see cref="AccessToken"/> renewal.
+        /// </remarks>
+        /// <param name="getTokenAsync">A delegate that returns a <see cref="ValueTask"/> of type <see cref="AccessToken"/>.
+        /// If <see cref="GetToken"/> is called, the provided delegate will be executed sync over async.</param>
+        /// <returns></returns>
+        public static TokenCredential Create(
+            Func<TokenRequestContext, CancellationToken, ValueTask<AccessToken>> getTokenAsync) => new StaticTokenCredential(getTokenAsync);
```

Comment:
Does the func need to return `ValueTask`? If not I wonder if `Task<AccessToken>` would be easier for users.

Reply:
I wanted to keep it 1:1 with GetToken's signature, unless you feel strongly otherwise

Reply:
I don't feel strongly I just wanted to pose the question to others.